[25.0] Prevent waiting for history item state to be ok in uploader

ahmedhamidawan opened the pull request:
Fixes https://github.com/galaxyproject/galaxy/issues/20241 _(discussion on the issue describes the reasoning here as well)_

We now wait for the extension to change from `auto` and prevent an item from being used if it gets a terminal state other than `ok` and `deferred`.

https://github.com/user-attachments/assets/71bbfc28-f572-44b0-9cb8-3d761608f96a

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
[25.0] Prevent waiting for history item state to be ok in uploader

Fixes https://github.com/galaxyproject/galaxy/issues/20241

We now wait for the extension to change from `auto` and prevent an item from being used if it gets a terminal state other than `ok` and `deferred`.

ahmedhamidawan committed:
add close button for split collection creator in workflow run

For the case where we have multiple collection type options for a collection input, the form data context button is not a toggle; as in it doesn't show/hide the creator. Therefore, added a close button to the side.

ahmedhamidawan committed:
remove wait for upload state alert

mvdbeek commented on lib/galaxy_test/selenium/test_workflow_run.py:
```diff
@@ -583,7 +583,6 @@ def _upload_hello_world_for_input(self, workflow_input, count=1, from_hid=1):
             workflow_input.title(n=i).wait_for_and_clear_and_send_keys(f"hello world.{i + 1}.fastq")
 
         workflow_input.embedded_start_button.wait_for_and_click()
-        workflow_input.status.wait_for_present()
```
To actually test this change I would remove `self.history_panel_wait_for_hid_ok(i)`

ahmedhamidawan committed:
No need to wait for item hids, waiting for the button itself is better

Co-authored-by: mvdbeek <m.vandenbeek@gmail.com>

ahmedhamidawan committed:
wait for add items button to enable; no need to switch to create tab

The test would fail because the add items to collection button's disabled/enabled state was not being detected, and hence, we would manually switch tabs and the items would not even have been added to the list yet. Now, we wait for the button to enable, and then adding the items automatically switches tab to Create.
